src/main/java/com/google/api/codegen/ruby/RubyGapicContext.java

```diff
@@ -358,6 +361,54 @@ public DynamicLangApiMethodView getMethodView(Interface service, Method method)
     return methodTransformer.generateDynamicLangApiMethod(methodContext);
   }
 
+  public List<GrpcStubView> getStubs(Interface service) {
+    ModelTypeTable modelTypeTable =
+        new ModelTypeTable(
+            new RubyTypeTable(getApiConfig().getPackageName()),
+            new RubyModelTypeNameConverter(getApiConfig().getPackageName()));
+    SurfaceTransformerContext context =
+        SurfaceTransformerContext.create(
+            service,
+            getApiConfig(),
+            modelTypeTable,
+            new RubySurfaceNamer(getApiConfig().getPackageName()));
+    return generateGrpcStubs(context);
+  }
+
+  private List<GrpcStubView> generateGrpcStubs(SurfaceTransformerContext context) {
```

Comment:
Instead of duplicating this method between Ruby & Node, create a GrpcStubTransformer under c.g.a.codegen.transformer that can be used by multiple languages, and call it from here.